lib/simp/cli/commands/environment/new.rb

```diff
@@ -0,0 +1,177 @@
+                options[:puppetfile] = true
+                # TODO: implement
+                fail NotImplementedError, 'TODO: implement --skeleton'
+              end
+
+      opts.on('--copy ENVIRONMENT', Simp::Cli::Utils::REGEXP_PUPPET_ENV_NAME,
+              'Copy assets from ENVIRONMENT') do |_src_env|
+                # TODO: implement
+                fail NotImplementedError, 'TODO: implement --copy'
+              end
+
+      opts.on('--link ENVIRONMENT', Simp::Cli::Utils::REGEXP_PUPPET_ENV_NAME,
+              'Symlink Secondary and Writeable environment directories',
+              'to ENVIRONMENT.  If --puppet-env is set, the Puppet',
+              'environment will --copy.') do |_src_env|
+                # TODO: implement
+                # TODO: implement --puppet-env => --copy logic
+                fail NotImplementedError, 'TODO: implement --link'
+              end
+
+      opts.on('--[no-]puppetfile',
+              'Generate Puppetfiles in Puppet env directory',
+              '  * `Puppetfile` will only be created if missing',
+              '  * `Puppetfile.simp` will be generated from RPM/',
+              '  * implies `--puppet-env`') do |v|
+        if (options[:types][:puppet][:puppetfile] = v)
+          options[:types][:puppet][:enabled] = true
+        end
+      end
+
+      opts.on('--[no-]puppetfile-install',
+              'Automatically deploys Puppetfile in Puppet environment',
+              'directory after creating it',
+              '  * implies `--puppet-env`',
+              '  * Does NOT imply `--puppetfile`') do |v|
+        if (options[:types][:puppet][:puppetfile_install] = v)
+          options[:types][:puppet][:enabled] = true
+        end
+      end
+
+      opts.on('--[no-]puppet-env',
+              'Includes Puppet environment when `--puppet-env`',
+              '(default: --no-puppet-env)') { |v| options[:types][:puppet][:enabled] = v }
+
+      opts.on('--[no-]secondary-env',
+              'Includes Secondary environment when `--secondary-env`',
+              '(default: --secondary-env)') { |v| options[:types][:secondary][:enabled] = v }
+
+      opts.on('--[no-]writable-env',
+              'Includes writable environment when `--writable-env`',
+              '(default: --writable-env)') { |v| options[:types][:writable][:enabled] = v }
+
+      opts.separator ''
+      opts.on_tail('-h', '--help', 'Print this message') do
+        puts opts
+        exit
+      end
+    end
+    opt_parser.parse!(args)
+    options
+  end
+
+  # Run command logic
+  # @param args [Array<String>] ARGV-style args array
+  def run(args)
+    options = parse_command_line(args)
+    action  = options.delete(:action)
+
+    if args.empty?
+      warn('', '-' * 80, 'WARNING: \'ENVIRONMENT\' is required.', '-' * 80, '')
+      sleep 1
+      help
+    end
+
+    env = args.shift
+
+    unless env =~ Simp::Cli::Utils::REGEXP_PUPPET_ENV_NAME
+      fail("ERROR: '#{env}' is not an acceptable environment name")
```

Comment:
Please raise Simp::Cli::ProcessingError an an error derived from that error.  Otherwise, a backtrace will be spewed when this failure is encountered.

Reply:
:ballot_box_with_check: